frontend/src/components/SystemsTable/SystemTableTools.tsx

```diff
@@ -167,6 +182,11 @@ export function SystemTableTools({
       tooltipMessage =
         "Cannot perform multiple analysis on systems with different dataset names.";
     }
+    if (duplicateSelectedSystemNames) {
```

Comment:
Would you mind doing a little refactoring of our old code? Sorry to put you through this 😅! I don't remember adding the `>=3` condition myself, but the only difference between `selectedSystemIDs.length == 2` and `>=3` seems to be the button name (Pairwise Analysis vs Multiple System Analysis). The rest of the logic could be reused.

Reply:
Thanks for pointing it out.